omnipaxos_core/src/sequence_paxos.rs

```diff
@@ -220,16 +224,23 @@ where
     }
 
     fn handle_compaction(&mut self, c: Compaction) {
-        let decided_idx = self.storage.get_decided_idx();
-        let compacted_idx = self.storage.get_compacted_idx();
+        let decided_idx = self.storage.get_decided_idx().expect("Failed to get decided index");
+        let compacted_idx = self.storage.get_compacted_idx().expect("Failed to get compacted index");
         match c {
             Compaction::Trim(Some(idx)) if idx <= decided_idx && idx > compacted_idx => {
-                self.storage.trim(idx - compacted_idx);
-                self.storage.set_compacted_idx(idx);
+                self.storage.set_compacted_idx(idx).expect("Failed to set compacted index");
+                match self.storage.trim(idx - compacted_idx) {
+                    Ok(_) => self.cached_state.set_compacted_idx(idx),
+                    Err(_) => {
+                        self.storage.set_compacted_idx(self.cached_state.get_compacted_idx()).expect("Failed to restore compacted index");
+                        panic!();
```

Comment:
As I have pointed out before, it doesn't make sense to just panic or unwrap without any error message. Please add that.